crates/task-impls/src/consensus/mod.rs

```diff
@@ -958,15 +958,17 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
                     );
 
                     if let Err(e) = self.publish_proposal(view, event_stream).await {
-                        warn!("Failed to propose; error = {e:?}");
+                        error!("Failed to propose; error = {e:?}");
                     };
                 }
             }
             HotShotEvent::QuorumVoteSend(vote) => {
                 let Some(proposal) = &self.current_proposal else {
                     return;
                 };
-                if proposal.get_view_number() == vote.get_view_number() {
+                if proposal.get_view_number() == vote.get_view_number()
+                    && self.public_key != self.quorum_membership.get_leader(vote.get_view_number())
```

Comment:
Why can't we reset `current_proposal` if we are the leader?

Reply:
We can reset, I thought we needed it to propose and that we might be setting it to none too soon, but we don't need it to propose.